### PR TITLE
Markdown extensions reference

### PR DESCRIPTION
**Pull request by Blind4Basics**
I tried to go for the `references` way but... not sure I succeeded x)

* bright side: keeping it that way could spare the tutorial
* down side: too much examples? :/

I created the sidebar for those markdown files, but it's not showing up in the preview, for now... I'll check again and edit there.

EDIT: \o/ sidebar working!

### CODE REVIEW

File: content/references/markdown/extensions.md [hunk 1]
```diff
@@ -76,14 +109,14 @@ Hidden _if_ the active language is Ruby.
...
 
 > Hidden _if_ the active language is JavaScript **or** TypeScript.
 
-### Using Code Blocks Inside
 
-To use code blocks within these conditional blocks,
-use tildes (`~`) to open the conditional block or increase the number of backticks used.
+### Using Code Blocks Inside Conditional Blocks
+
+To use code blocks within these conditional blocks, use tildes (`~`) to declare the conditional block or increase the number of backticks used.
```
[the remark this answers is not in the document]
Comment (Blind4Basics) [reply content/references/markdown/extensions.md:117]: dunno, For a "reference", it seems interesting to me to give the information. But nothing imperative about that, yes. I remove it, so?

[the remark this answers is not in the document]
Comment (kazk) [reply content/references/markdown/extensions.md:117]: Hmm, maybe we can just link to this section from

> * You can use any markdown formatting you want inside conditional blocks.

by adding some warning about using backtick code fences.

File: content/references/markdown/extensions.md [hunk 2]
```diff
@@ -26,29 +40,48 @@ end
...
+If the selected language is java, this block is showing up in the description, rendered to:
+
+> For java, use `Preloaded.check(input)`.
+
+### Details
 
-These are useful when adding language specific notes.
+* The language name in the info string has to match a valid language ID. To find the language ID, visit its documentation from the [languages page](/languages/).
+* To obtain proper rendering, language specific blocks need an empty line before and after them, just like code blocks.
+* You can use any markdown formatting you want inside conditional blocks.
```
Comment (Blind4Basics) [inline content/references/markdown/extensions.md:74]:
```suggestion
* You can use any markdown formatting you want inside conditional blocks, but be careful [when inserting code fences](/references/markdown/extensions/#using-code-blocks-inside-conditional-blocks): prefere to use different kinds of code fences to distinguish them more easily.
```

File: content/references/markdown/extensions.md [hunk 3]
```diff
@@ -26,29 +40,48 @@ end
...
+If the selected language is java, this block is showing up in the description, rendered to:
+
+> For java, use `Preloaded.check(input)`.
+
+### Details
 
-These are useful when adding language specific notes.
+* The language name in the info string has to match a valid language ID. To find the language ID, visit its documentation from the [languages page](/languages/).
+* To obtain proper rendering, language specific blocks need an empty line before and after them, just like code blocks.
+* You can use any markdown formatting you want inside conditional blocks, but be careful [when inserting code fences](/references/markdown/extensions/#using-code-blocks-inside-conditional-blocks): prefere to use different kinds of code fences to distinguish them more easily.
```
Comment (Blind4Basics) [inline content/references/markdown/extensions.md:74]:
```suggestion
* You can use any markdown formatting you want inside conditional blocks, but be careful when inserting code blocks: prefere to use different kinds of code fences to distinguish them more easily (like `~~~` and ```` ``` ````: [see here](/references/markdown/extensions/#using-code-blocks-inside-conditional-blocks)).
```

File: content/references/markdown/extensions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Blind4Basics) [reply content/references/markdown/extensions.md:117]: * added note in the bullet point above
* didn't change the section about "increasing the number of backticks"

File: content/references/markdown/extensions.md [hunk 4]
```diff
@@ -26,29 +40,48 @@ end
...
+If the selected language is java, this block is showing up in the description, rendered to:
+
+> For java, use `Preloaded.check(input)`.
+
+### Details
 
-These are useful when adding language specific notes.
+* The language name in the info string has to match a valid language ID. To find the language ID, visit its documentation from the [languages page](/languages/).
+* To obtain proper rendering, language specific blocks need an empty line before and after them, just like code blocks.
+* You can use any markdown formatting you want inside conditional blocks, but be careful when inserting code blocks: prefere to use different kinds of code fences to distinguish them more easily (like `~~~` and ```` ``` ````: [see here](/references/markdown/extensions/#using-code-blocks-inside-conditional-blocks)).
```
Comment (kazk) [inline content/references/markdown/extensions.md:74]:
```suggestion
* You can use Markdown inside conditional blocks, but [be careful when inserting code blocks](/references/markdown/extensions/#using-code-blocks-inside-conditional-blocks).
```

**Pull request merged**